test/api/unit_roles_test.rb

```diff
@@ -46,4 +56,23 @@ def test_post_unit_roles_not_unique
     assert_equal to_post[:unit_id], last_response_body['unit_id']
     assert_equal to_post[:user_id], last_response_body['user_id']
   end
+
+  # Delete unit role
+  def test_unit_roles_delete
+    number_of_unit_roles = UnitRole.all.count
+
+    test_unit_role = UnitRole.all.first
+    id_of_unit_role_to_delete = test_unit_role.id
+
+    # perform the delete
+    delete_json with_auth_token "/api/unit_roles/#{id_of_unit_role_to_delete}"
+
+    # Check there is one less unit_roles
+    assert_equal number_of_unit_roles - 1, UnitRole.all.count
+
+    # Check that you can't find the deleted id
+    refute UnitRole.exists?(id_of_unit_role_to_delete)
+    assert_equal last_response.status, 200
+  end
```

Comment:
This is better - would be good to also test this with an account that is not permitted to delete the unit role.